### PR TITLE
OLS-778, OLS-514: Update data collector configuration

### PR DESCRIPTION
**Pull request by onmete**
## Description

Update data collector configuration according to these changes: https://github.com/openshift/lightspeed-service/pull/1364

## Type of change

- [x] Refactor
- [x] New feature

### CODE REVIEW

**Comment by raptorsun** [comment]
the logic looks good, just need to regenerate the bundle so that we have the new CRD in the bundle:
1. update the bundle using this script
```shell
./hack/update_bundle_catalog.sh
```
2. revert changes in files other than  the CSV file `bundle/manifests/lightspeed-operator.clusterserviceversion.yaml`
3. revert the change in image references in the CSV file `bundle/manifests/lightspeed-operator.clusterserviceversion.yaml`

**Comment by onmete** [comment]
@raptorsun done

File: bundle/manifests/lightspeed-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -37,12 +37,12 @@ metadata:
...
     capabilities: Basic Install
-    createdAt: "2024-07-27T05:27:04Z"
+    createdAt: "2024-08-22T09:58:45Z"
     features.operators.openshift.io/cnf: "false"
     features.operators.openshift.io/cni: "false"
     features.operators.openshift.io/csi: "false"
-    features.operators.openshift.io/disconnected: "false"
-    features.operators.openshift.io/fips-compliant: "false"
+    features.operators.openshift.io/disconnected: "true"
+    features.operators.openshift.io/fips-compliant: "true"
```
Comment (raptorsun) [inline bundle/manifests/lightspeed-operator.clusterserviceversion.yaml:45]: we need to revert these 2 properties to `false`.

**Comment by raptorsun** [comment]
/override "Red Hat Konflux / ols-e2e-tests-rosa / bundle"
/override "Red Hat Konflux / ols-e2e-tests-rosa / lightspeed-operator"
/override "Red Hat Konflux / ols-enterprise-contract / bundle"
/override "Red Hat Konflux / ols-enterprise-contract / lightspeed-operator"

**Pull request merged**